openssl-dynamic/src/main/c/ssl.c

```diff
@@ -732,43 +686,30 @@ static void ssl_dyn_lock_function(int mode, struct CRYPTO_dynlock_value *l,
 static void ssl_dyn_destroy_function(struct CRYPTO_dynlock_value *l,
                           const char *file, int line)
 {
-    apr_status_t rv;
-    rv = apr_thread_mutex_destroy(l->mutex);
-    if (rv != APR_SUCCESS) {
-        /* TODO log that fprintf(stderr, "Failed to destroy mutex for dynamic lock %s:%d", l->file, l->line); */
-    }
+    free((void *) l->file);
```

Comment:
is the cast necessary? can we just make the type `char*` in the structure definition if the `const` is confusing things?